symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -0,0 +1,148 @@
+package com.symphony.bdk.core.service.datafeed.impl;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.gen.api.model.V4Event;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Slf4j
+public class DatafeedServiceV1 extends AbstractDatafeedService {
+
+    private static final String DATAFEED_ID_FILE = "datafeed.id";
+    private final AtomicBoolean started = new AtomicBoolean();
+    private String datafeedId;
+
+
+    public DatafeedServiceV1(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, podClient, authSession, config);
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() throws Throwable {
+        log.debug("Start retrieving datafeed id from disk");
+        this.datafeedId = this.retrieveDatafeedIdFromDisk();
+
+        if (this.datafeedId == null) {
+            log.debug("Start creating a new datafeed and save to disk");
+            this.datafeedId = this.createDatafeedAndSaveToDisk();
+        }
+
+        log.debug("Start reading events from datafeed {}", datafeedId);
+        do {
+            this.started.set(true);
+            this.readDatafeed();
+        } while (this.started.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    private void readDatafeed() throws Throwable {
+        this.retry.executeCheckedSupplier(() -> {
+            try {
+                List<V4Event> events = datafeedApi.v4DatafeedIdReadGet(datafeedId, authSession.getSessionToken(), authSession.getKeyManagerToken(), null);
+                if (events != null && !events.isEmpty()) {
+                    handleV4EventList(events);
+                }
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    try {
```

Comment:
Do we need a try catch just to log and throw the very same exception ? Exception with stack trace will be shown in the logs, won't it?